lib/pdk/cli/release.rb

```diff
@@ -0,0 +1,295 @@
+        end
+      else
+        new_version = 'v0.0.1'
+        changelog_generator = PDK::CLI::Util::ChangelogGenerator.new
+        if opts[:version_number]
+          new_version = opts[:version_number]
+        else
+          # Use changelog generator to establish version
+          PDK.logger.info _('Generating changelog to get next version number')
+          changelog_generator.generate_changelog
+
+          new_version = changelog_generator.get_next_version(module_metadata.data['version'])
+        end
+
+        PDK.logger.info _('Updating version to %{module_version}') % {
+          module_version: new_version,
+        }
+
+        # Bump version in metadata file
+
+        module_metadata.data['version'] = new_version
+        module_metadata.write!('metadata.json')
+
+        # Create new changelog with the correct version
+        changelog_generator.generate_changelog
+        PDK.logger.info _("New changelog generated with version #{new_version}")
+      end
+
+      if opts[:skip_documentation]
+        PDK.logger.info _('Skipping documentation update for this module')
+      else
+        PDK.logger.info _('Updating documentation using puppet strings')
+        result = system('bundle exec puppet strings generate --format markdown --out REFERENCE.md')
+        unless result
+          PDK.logger.error _('Error updating documentation with puppet strings.')
+        end
+      end
+
+      if opts[:skip_dependency]
+        PDK.logger.info _('Skipping dependency-cheking on the metadata of this module')
+      else
+        # run dependency-checker and output dependent modules list
+        PDK.logger.info _('Running dependency checker for %{module_name} - version %{module_version}') % {
+          module_name:    module_metadata.data['name'],
+          module_version: module_metadata.data['version'],
+        }
+
+        PDK.logger.warn _('Dependency-checker failed to execute, please make sure the gem is installed') unless system('dependency-checker metadata.json')
+      end
+
+      # TODO: Ensure forge metadata has been set, or call out to interview
+      #       to set it.
+      #
+      unless module_metadata.forge_ready?
+        if opts[:force]
+          PDK.logger.warn _(
+            'This module is missing the following fields in the metadata.json: %{fields}. ' \
+            'These missing fields may affect the visibility of the module on the Forge.',
+          ) % {
+            fields: module_metadata.missing_fields.join(', '),
+          }
+        else
+          module_metadata.interview_for_forge!
+          module_metadata.write!('metadata.json')
+        end
+      end
+
+      builder = PDK::Module::Build.new(opts)
+
+      unless opts[:force]
+        if builder.package_already_exists?
+          PDK.logger.info _("The file '%{package}' already exists.") % { package: builder.package_file }
+
+          unless PDK::CLI::Util.prompt_for_yes(_('Overwrite?'), default: false)
```

Comment:
Should there be an option to ignore the overwrite? Only way to avoid this is call --force which has other implications

Reply:
The easiest method is to use force for this. As this command is meant to be interactive I don't think it should be problematic.